app/services/form_copy_service.rb

```diff
@@ -0,0 +1,101 @@
+class FormCopyService
+  TO_INCLUDE = %i[creator_id created_at updated_at creator_id].freeze
+  TO_EXCLUDE = Form::ATTRIBUTES_NOT_IN_FORM_DOCUMENT + TO_INCLUDE
+
+  def initialize(form)
+    @form = form
+    @copied_form = Form.new
+  end
+
+  def copy(tag: "draft")
+    form_doc = FormDocument.find_by(form_id: @form.id, tag:, language: @form.language)
+    return false if form_doc.blank?
+
+    content = form_doc.content
+
+    ActiveRecord::Base.transaction do
+      copy_attributes(content)
+      prepend_name
+
+      @copied_form.save!
+      copy_pages(content["steps"])
+      copy_routing_conditions(content["steps"])
+      copy_group
+
+      @copied_form
```

Comment:
In testing, I found that copying a form and trying to preview pages didn't work. The draft FormDocument didn't have any steps.

I think this is because the draft FormDocument is created when the form is saved.

I think it can be fixed by moving the `@copied_form.save` to line 24, after the pages have been added.

That way the when the draft is created, it will include the pages?